### ISSUE TITLE
NullPointerException from KotlinMetadata.isMissingSyntheticAnnotationMethod

### ISSUE THREAD

**Issue by plnice** [google/dagger#1728]
Hi, after bumping Dagger to 2.26 we see following exception thrown from annotation processor in one of our modules:
```
Caused by: java.lang.NullPointerException
        at dagger.internal.codegen.kotlin.KotlinMetadata.lambda$new$0(KotlinMetadata.java:89)
        at dagger.internal.codegen.kotlin.KotlinMetadata.lambda$new$2(KotlinMetadata.java:90)
        at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:167)
        at dagger.internal.codegen.kotlin.KotlinMetadata.isMissingSyntheticAnnotationMethod(KotlinMetadata.java:158)
        at dagger.internal.codegen.kotlin.KotlinMetadataUtil.isMissingSyntheticPropertyForAnnotations(KotlinMetadataUtil.java:76)
        at dagger.internal.codegen.validation.DependencyRequestValidator.checkQualifiers(DependencyRequestValidator.java:74)
        at dagger.internal.codegen.validation.DependencyRequestValidator.validateDependencyRequest(DependencyRequestValidator.java:67)
        at dagger.internal.codegen.validation.InjectValidator.validateDependencyRequest(InjectValidator.java:244)
        at dagger.internal.codegen.validation.InjectValidator.validateField(InjectValidator.java:205)
        at dagger.internal.codegen.validation.InjectValidator.validateMembersInjectionType(InjectValidator.java:256)
        at dagger.internal.codegen.validation.InjectBindingRegistryImpl.tryRegisterMembersInjectedType(InjectBindingRegistryImpl.java:280)
        at dagger.internal.codegen.validation.InjectBindingRegistryImpl.tryRegisterMembersInjectedType(InjectBindingRegistryImpl.java:264)
        at dagger.internal.codegen.InjectProcessingStep$1.visitVariableAsField(InjectProcessingStep.java:54)
        at dagger.internal.codegen.InjectProcessingStep$1.visitVariableAsField(InjectProcessingStep.java:44)
        at com.sun.tools.javac.code.Symbol$VarSymbol.accept(Symbol.java:1237)
        at dagger.internal.codegen.InjectProcessingStep.process(InjectProcessingStep.java:76)
        at dagger.internal.codegen.validation.TypeCheckingProcessingStep.lambda$process$0(TypeCheckingProcessingStep.java:51)
        at dagger.internal.codegen.validation.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:48)
        at dagger.internal.codegen.validation.TypeCheckingProcessingStep.process(TypeCheckingProcessingStep.java:34)
        at dagger.internal.codegen.statistics.DaggerStatisticsCollectingProcessingStep.process(DaggerStatisticsCollectingProcessingStep.java:52)
        at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:330)
        at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:181)
        at org.jetbrains.kotlin.kapt3.base.incremental.IncrementalProcessor.process(incrementalProcessors.kt)
```
any idea what could have cause it? Can we find out which part of our code is this related to?

**Comment by danysantiago** [comment]
Hey - This is occurring while validating a member injected field in a Kotlin source. i.e. something like `@Inject lateinit var data: Data`. What version of Kotlin are you on? Which version of Dagger where you using before? Does any of those member injected fields stand out? As in, do they have some other annotation, like a qualifier, or you have some get() or set() logic in them?

**Comment by plnice** [comment]
After some more investigation, it turned out that the processing failed for the first class being processed and the `fieldSignature` for all the (regular) fields in that class were `null`. 

The problem was due to conflict with Room annotation processor when `dagger-compiler` is applied only for the specific build type and the room processor is applied for all build types:
```
kaptRelease <dagger-compiler>
kapt <room-compiler>
```

The issue was not related to any particular Dagger version (in the same changeset I was also adding dagger-reflect which caused the change in applying dagger-compiler in Gradle).

If anyone also runs into the same problem, I managed to workaround this by specifying room-compiler for all the build types separately:
```
kaptRelease <dagger-compiler>
kaptDebug <room-compiler>
kaptRelease <room-compiler>
```

The issue is probably related to the order of applied processors and room-compiler doing some nasty things, but didn't investigate it more after finding a workaround.
